gemelli/log_ratios.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+np.seterr(divide='ignore')
+
+
+def log_ratio(numerator, denominator, pseudocount=None):
+    """
+    Take the log-ratio of the numerator and the denominator.
+
+    Parameters
+    ----------
+    numerator : array_like, float
+    denominator : array_like, float
+    pseudocount : float, optional
+
+    Returns
+    ----------
+    log-ratio: array_like, float
+        log ratio of numerator and denominator
+
+    Raises
+    ----------
+    ValueError
+        If shape numerator and denominator are not equal.
+    ValueError
+        If there is a pseudocount and the log of the numerator or
+        denominator contains non-finite values
+
+    Examples
+    ----------
+    TODO
+
+    """
+
+    numerator = np.atleast_1d(numerator)
+    denominator = np.atleast_1d(denominator)
+
+    if numerator.shape != denominator.shape:
+        raise ValueError('The numerator and denominator must be of',
+                         ' equal length.')
+
+    if pseudocount is not None:
+        numerator = np.log(numerator + pseudocount)
+        denominator = np.log(denominator + pseudocount)
+        # ensure log-ratio has no zeros
+        # this only a problem in
+        if not all(np.isfinite(numerator)):
+            raise ValueError('Log of numerator is non-finite')
+        if not all(np.isfinite(denominator)):
+            raise ValueError('Log of denominator is non-finite')
+    else:
+        numerator = np.log(numerator)
+        denominator = np.log(denominator)
+
+    return numerator - denominator
+
+
+def percentile_ratio(mat, ranks, pseudocount=None,
+                     percent=25, interpolation='midpoint'):
+    """
+    Sum samples of count matrix with the upper and lower percentile of
+    ranked features and then perform a log-ratio between them.
+
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of counts where
+       rows = features and
+       columns = samples
+    ranks : array_like, float
+        a matrix of feature ranks
+    pseudocount: float, optional
+        Add a pseudocount before taking the log-ratio. Default is None.
+    percent: int, optional [1,100]
+        The upper and lower percentile of ranks for the log-ratio.
+        The default is quartiles (i.e. 25).
```

Comment:
I think the `percent` part of this docstring needs work. I think it should be `optional[1, 50]` and described as an interval size of the quantiles, at least based on the code below

Reply:
yes replaced.